### PR TITLE
Update ArduinoOTA.cpp to solve #4086

### PR DESCRIPTION
**Pull request by lrmoreno007**
Update ArduinoOTA.cpp to solve #4086

### CODE REVIEW

File: libraries/ArduinoOTA/ArduinoOTA.cpp [hunk 1]
```diff
@@ -146,26 +146,26 @@ int ArduinoOTAClass::parseInt(){
...
-    data[index++] = _udp_ota->read();
+    data[index] = _udp_ota->read();
   }
   return 0;
 }
 
 String ArduinoOTAClass::readStringUntil(char end){
   String res = "";
-  int value;
+  char value;
```
Comment (igrr) [inline libraries/ArduinoOTA/ArduinoOTA.cpp]: What is the motivation for this change?

File: libraries/ArduinoOTA/ArduinoOTA.cpp [hunk 2]
```diff
@@ -146,26 +146,26 @@ int ArduinoOTAClass::parseInt(){
   uint8_t index;
   char value;
   while(_udp_ota->peek() == ' ') _udp_ota->read();
-  for(index = 0; index < sizeof(data); ++index){
+  for(index = 0; index < sizeof(data); index++){
     value = _udp_ota->peek();
     if(value < '0' || value > '9'){
       data[index++] = '\0';
```
Comment (Juppit) [inline libraries/ArduinoOTA/ArduinoOTA.cpp]: For me, it would be even clearer to not incrementieren the index here, something like this:
`data[index+1] = '\0';`

File: libraries/ArduinoOTA/ArduinoOTA.cpp [hunk 3]
```diff
@@ -146,26 +146,26 @@ int ArduinoOTAClass::parseInt(){
   uint8_t index;
   char value;
   while(_udp_ota->peek() == ' ') _udp_ota->read();
-  for(index = 0; index < sizeof(data); ++index){
+  for(index = 0; index < sizeof(data); index++){
```
Comment (devyte) [inline libraries/ArduinoOTA/ArduinoOTA.cpp]: This change is undesired (++index is better style in C++ in a for loop)

File: libraries/ArduinoOTA/ArduinoOTA.cpp [hunk 4]
```diff
@@ -146,26 +146,26 @@ int ArduinoOTAClass::parseInt(){
   uint8_t index;
   char value;
```
Comment (devyte) [inline libraries/ArduinoOTA/ArduinoOTA.cpp]: value should be declared int, because peek() returns an int that can be -1 on error

File: libraries/ArduinoOTA/ArduinoOTA.cpp [hunk 5]
```diff
@@ -146,26 +146,26 @@ int ArduinoOTAClass::parseInt(){
...
   while(_udp_ota->peek() == ' ') _udp_ota->read();
-  for(index = 0; index < sizeof(data); ++index){
+  for(index = 0; index < sizeof(data); index++){
     value = _udp_ota->peek();
     if(value < '0' || value > '9'){
```
Comment (devyte) [inline libraries/ArduinoOTA/ArduinoOTA.cpp]: This if statement needs a check against peek error return: if(value < 0 || everythingelse)

File: libraries/ArduinoOTA/ArduinoOTA.cpp (hunk 1 above)
Comment (devyte) [reply libraries/ArduinoOTA/ArduinoOTA.cpp]: value should be declared int, because read() returns an int that can be -1 on error

File: libraries/ArduinoOTA/ArduinoOTA.cpp [hunk 6]
```diff
@@ -146,26 +146,26 @@ int ArduinoOTAClass::parseInt(){
...
 }
 
 String ArduinoOTAClass::readStringUntil(char end){
   String res = "";
-  int value;
+  char value;
   while(true){
     value = _udp_ota->read();
-    if(value < 0 || value == '\0' || value == end){
+    if(value < '0' || value == '\0' || value == end){
```
Comment (devyte) [inline libraries/ArduinoOTA/ArduinoOTA.cpp]: This change needs to be undon. The check value < 0 was meant to cover an error return from peek() or read(). Undo.

File: libraries/ArduinoOTA/ArduinoOTA.cpp [hunk 7]
```diff
@@ -146,26 +146,26 @@ int ArduinoOTAClass::parseInt(){
...
       return res;
     }
-    res += static_cast<char>(value);
+    res += value;
```
Comment (devyte) [inline libraries/ArduinoOTA/ArduinoOTA.cpp]: This change needs to be undone, given that value needs to be an int.

File: libraries/ArduinoOTA/ArduinoOTA.cpp (hunk 2 above)
Comment (devyte) [reply libraries/ArduinoOTA/ArduinoOTA.cpp]: This should be data[index] = '\0';

File: libraries/ArduinoOTA/ArduinoOTA.cpp (hunk 1 above)
Comment (lrmoreno007) [reply libraries/ArduinoOTA/ArduinoOTA.cpp]: The motivation is back to original before yoursunny PR (it was char) and homogenize data types ( '/0' and end are char)
